### PR TITLE
Next.js compliant command for production builds

### PR DESCRIPTION
**Pull request by omnus**
## Description and Related Issue(s)

This PR creates a new command (`build:next`) to allow easy deployment to CI platforms like Vercel.

### Proposed Changes

Creation of a new command `build:next` that downloads external assets, builds both sprite.svg and envs.js and then calls next build. This allows for a simple production deployment that includes the files needed for the frontend to run.

### Breaking or Incompatible Changes

* None, existing scripts and commands are unchanged.

### Additional Information

To deploy using this new command you need to override the build command from `next build` (or equivalent) to use the `build:next` command (e.g. `yarn build:next`).

## Checklist for PR author
- ✅ I have tested these changes locally.
- ✅ I added tests to cover any new functionality, following this [guide]
- ✅ Whenever I fix a bug, I include a regression test to ensure that the bug does not reappear silently.
- [n/a] If I have added, changed, renamed, or removed an environment variable
    - I updated the list of environment variables in the [documentation](ENVS.md) 
    - I made the necessary changes to the validator script according to the [guide](./CONTRIBUTING.md#adding-new-env-variable)
    - I added "ENVs" label to this pull request

### CODE REVIEW

**Comment by omnus** [comment]
Hi @tom2drum and @isstuev! 

Requesting a review on this PR please. It's just a small one, but I think it could be useful for people looking to deploy the frontend standalone. 

Thanks! 🫡

File: package.json [hunk 1]
```diff
@@ -12,6 +12,7 @@
     "dev:preset": "./tools/scripts/dev.preset.sh",
     "dev:preset:sync": "tsc -p ./tools/preset-sync/tsconfig.json && node ./tools/preset-sync/index.js",
     "build": "next build",
+    "build:vercel": "./deploy/scripts/build_assets.sh && next build",
```
Comment (tom2drum) [inline package.json:15]: I would argue against mentioning a particular CI vendor in the command name. The command is performing a generic task, so we should not be that specific when naming it. I suggest changing it to something like: `build:next`, `build:app`, `build:machine`, `build:node`, etc., or modify the build script of `package.json` itself. The last option implies that its usage also should be modified.

File: deploy/scripts/build_assets.sh [hunk 2]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+
+# generate sprite
+yarn svg:build-sprite
+echo ""
+
+# generate envs.js file
+dotenv \
+  -- bash -c './deploy/scripts/make_envs_script.sh' |
+pino-pretty
```
[the remark this answers is not in the document]
Comment (tom2drum) [reply deploy/scripts/build_assets.sh:10]: If no ENV variables should be loaded additionally and passed to the script, there is no need to use `dotenv` command.

So, overall code can be simplified to oneliner in `package.json` file:
```
  "build:vercel": "yarn svg:build-sprite && ./deploy/scripts/make_envs_script.sh && next build",
```

**Review by tom2drum — CHANGES_REQUESTED** [review]

File: deploy/scripts/build_assets.sh (hunk 2 above)
[the remark this answers is not in the document]
Comment (omnus) [reply deploy/scripts/build_assets.sh]: Great idea, done!

It also occurred to me that we should include the `download_assets.sh` script as well, so I have added that to the oneliner in package.json.

File: package.json (hunk 1 above)
Comment (omnus) [reply package.json]: Yup, I agree. I have done with `build:next` as that seems to be suitably general.

Thank you for the feedback!

**Comment by omnus** [comment]
Thanks for the ideas @tom2drum, I've implemented both of those as you suggested 🙏.

**Review by tom2drum — APPROVED** [review]

**Comment by tom2drum** [comment]
@omnus Nice job! Thanks 🙌

**Pull request merged**